mesonbuild/dependencies/misc.py

```diff
@@ -537,7 +537,7 @@ def shaderc_factory(env: 'Environment',
         shared_libs = ['shaderc']
         static_libs = ['shaderc_combined', 'shaderc_static']
 
-        if kwargs.get('static', False):
+        if kwargs.get('static', env.coredata.get_option(mesonlib.OptionKey('prefer_static'))):
```

Comment:
This, however, is a function, not a class, so kwargs.get() is correct as is your modification.